src/test/java/org/galatea/starter/entrypoint/IexRestControllerTest.java

```diff
@@ -81,4 +82,60 @@ public void testGetLastTradedPriceEmpty() throws Exception {
         .andExpect(jsonPath("$", is(Collections.emptyList())))
         .andReturn();
   }
+
+  @Test
+  public void testGetHistoricalPricesDate() throws Exception {
+
+    MvcResult result = this.mvc.perform(
+        org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+            .get("/iex/historicalPrices?symbol=FB&range=date&date=20211018")
+            .accept(MediaType.APPLICATION_JSON_VALUE))
+        .andExpect(status().isOk())
+        .andExpect(jsonPath("$[0].symbol", is("FB")))
+        .andExpect(jsonPath("$[0].close").value(new BigDecimal("335.34")))
+        .andExpect(jsonPath("$[0].high").value(new BigDecimal("335.89")))
+        .andExpect(jsonPath("$[0].low").value(new BigDecimal("327.5")))
+        .andExpect(jsonPath("$[0].open").value(new BigDecimal("328.95")))
+        .andExpect(jsonPath("$[0].volume").value(21585018))
+        .andExpect(jsonPath("$[0].date").value("2021-10-18"))
+        .andReturn();
+  }
+
+  @Test
+  public void testGetHistoricalPricesEmptySymbol() throws Exception {
```

Comment:
It'd be very rare to see empty request params, but good thing to check for. In general the 3 cases I'm always looking for are for each property that is being "used":
1) Null values
2) Invalid values
3) Valid values

Additionally, with a new data provider, a good test case to add would be if the external server request failed with say HTTP 500

Reply:
Added more tests